web_edit_user_filter/static/src/js/customFavorite.esm.js

```diff
@@ -0,0 +1,54 @@
+/** @odoo-module **/
+
+import {CustomFavoriteItem} from "@web/search/favorite_menu/custom_favorite_item";
+import {patch} from "@web/core/utils/patch";
+
+patch(CustomFavoriteItem.prototype, "web_edit_user_filter.CustomFavoriteItem", {
+    /**
+     * @param {Event} ev
+     */
+    saveFavorite(ev) {
+        if (!this.state.description) {
+            this.notificationService.add(
+                this.env._t("A name for your favorite filter is required."),
+                {type: "danger"}
+            );
+            ev.stopPropagation();
+            return this.descriptionRef.el.focus();
+        }
+        var searchItems = this.env.searchModel.getSearchItems(
+            (s) => s.type === "favorite"
+        );
+        const favorites = searchItems.filter(
+            (f) => f.description === this.state.description
+        );
```

Comment:
**suggestion**:

```suggestion
        var searchItems = this.env.searchModel.getSearchItems(
            (s) => s.type === "favorite" && s.description === this.state.description
        );
```

(there are a few similar simplifications that could be made around the code)

Reply:
Modified as suggested. Identified in few other places and modified it. However, could not modify in places like [here](https://github.com/OCA/web/pull/3007/files#diff-6d0be0c9045ebd4c2ce6f7e9cac9d59acdcf7491d10856a32f5b6dd1f546f03bR26) and popover is not visible when modifying it.